pilot/pkg/xds/ads.go

```diff
@@ -144,13 +144,10 @@ func isExpectedGRPCError(err error) bool {
 func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.DiscoveryRequest, errP *error) {
 	defer func() {
 		close(reqChannel)
-		// Close the initialized channel, if its not already closed, to prevent blocking the stream
-		select {
-		case <-con.initialized:
-		default:
-			close(con.initialized)
-		}
+		// Close the stop channel, to prevent blocking the stream
+		con.Stop()
```

Comment:
need to be careful to not close it 2x in the future. for now it looks good though